WordPress/gutenberg
# test: Add script to start local Appium server

## What?

Simplify connecting the Appium Inspector to the mobile Demo editor,
without the need to run the e2e test suite. This is useful for identifying
selector strategies for elements while writing tests.

## Why?

Getting a device or simulator/emulator to the correct state where the app is
installed, WDA is installed, and an Appium server is actively running can be
difficult. The e2e testing scripts automate setting all of those up when
running e2e tests, but it is difficult to inspect the app while automated tests
are actively running.

Adding this utility will help contributors avoid unexpected errors like the
following:

```
Error
Could not connect to server; are you sure it's running? If you are using the browser version, also ensure your Appium server has been started with --allow-cors.
```

```
Call to 'getPageSource' failed
Cannot read property 'client' of null
```

## How?

Add native editor script to start a local Appium server for debugging.

## Testing Instructions




1. Set up your local environment to use Xcode 13, e.g.
   `sudo xcode-select -s /Applications/Xcode\ 13.app`.
1. Find your desired testing device ID: `xcrun simctl list`
1. Boot the device: `xcrun simctl boot <device-id>`
1. Install the Pods: `npm run native preios`
1. Start the development server: `npm run native start:reset`
1. Press <kbd>i</kbd> to build and run the app on the open iOS simulator.
1. Install WDA on the device: `npm run test:e2e:build-wda`
1. Start the Appium server: `npm run native appium:start`
1. Launch the Appium Inspector app, configure the inspector for the device by
   mirroring the [capabilities](https://github.com/WordPress/gutenberg/blob/30b2659aef227dc136d641543e653482c16df77f/packages/react-native-editor/__device-tests__/helpers/caps.js) configuration in the project.
1. Verify you can inspect the app.

<details><summary>Example iPhone 13 capabilities</summary>

```json
{
  "platformName": "iOS",
  "appium:os": "iOS",
  "appium:deviceOrientation": "portrait",
  "appium:automationName": "XCUITest",
  "appium:deviceName": "iPhone 13",
  "appium:platformVersion": "15.5"
}
```

</details>

### Testing Instructions for Keyboard

n/a

## Screenshots or screencast 
n/a
